### PR TITLE
FIX create utils module

### PR DESCRIPTION
**Pull request by romanlutz**
This adds an `__init__.py` file just like #877 on the release branch, but additionally moves other utilities (input validation) into the module. The move requires a few import adjustments which are also included.

### CODE REVIEW

File: fairlearn/reductions/_moments/utility_parity.py [hunk 1]
```diff
@@ -6,7 +6,8 @@
 
 from .moment import ClassificationMoment
 from .moment import _GROUP_ID, _LABEL, _PREDICTION, _ALL, _EVENT, _SIGN
-from fairlearn._input_validation import _MESSAGE_RATIO_NOT_IN_RANGE, _validate_and_reformat_input
+from fairlearn.utils._input_validation import _MESSAGE_RATIO_NOT_IN_RANGE, \
+    _validate_and_reformat_input
```
Comment (adrinjalali) [inline fairlearn/reductions/_moments/utility_parity.py:10]:
```suggestion
from fairlearn.utils._input_validation import (
    _MESSAGE_RATIO_NOT_IN_RANGE,
    _validate_and_reformat_input
)
```

Or even two separate imports :D

File: fairlearn/utils/__init__.py [hunk 2]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
```
Comment (adrinjalali) [inline fairlearn/utils/__init__.py:1]:
```suggestion
# Copyright (c) Fairlearn contributors.
```

maybe? didn't we change it to this?

File: test/unit/postprocessing/test_threshold_optimization.py [hunk 3]
```diff
@@ -7,7 +7,7 @@
 from copy import deepcopy
 import numpy as np
 import pytest
-from fairlearn._input_validation import \
+from fairlearn.utils._input_validation import \
     (_MESSAGE_Y_NONE,
```
Comment (adrinjalali) [inline test/unit/postprocessing/test_threshold_optimization.py:11]:
```suggestion
from fairlearn.utils._input_validation import (
    _MESSAGE_Y_NONE,
```

**Review by adrinjalali — APPROVED** [review]
Some nits which you can ignore if you don't like the style, otherwise LGTM.

I'd cherry pick this for the branch rather than merging the other PR.

File: fairlearn/utils/__init__.py (hunk 2 above)
Comment (hildeweerts) [reply fairlearn/utils/__init__.py:1]: According to Miro's comment [here](https://github.com/fairlearn/fairlearn/pull/823#issuecomment-857661650)  removing existing copyright from the source code might be in conflict with the MIT license.

Comment (adrinjalali) [reply fairlearn/utils/__init__.py:1]: Not if all contributors agree + this is new code with be license

Comment (hildeweerts) [reply fairlearn/utils/__init__.py:1]: Good points, in that case I'm very happy to change it to Fairlearn contributors :)

**Review by riedgar-ms — APPROVED** [review]

**Pull request merged**